chia/wallet/trade_manager.py

```diff
@@ -547,7 +551,11 @@ async def _create_offer_for_ids(
 
             all_transactions: List[TransactionRecord] = []
             fee_left_to_pay: uint64 = fee
-            for id, selected_coins in coins_to_offer.items():
+            # TODO: Create the transaction in place above when we select the coins? The access of the sorted keys here
+            #       makes sure we create the XCH transaction first to make sure we pay fee with the XCH side of the
+            #       offer and don't create an extra fee transaction in other wallets.
```

Comment:
You're not going to be able to create the transaction above.  Each transaction needs to assert an announcement that contains a hash of all the coins used in the entire offer.  This is why we must select the coins first -> calculate the bundle announcement -> create the transactions.  That part of this comment (assuming I read it correctly) should be deleted.